### PR TITLE
Decision record on JWT hardening

### PR DESCRIPTION
**Pull request by marcgs**
Decision record on JWT hardening.

See follow-up story: https://github.com/agera-edc/MinimumViableDataspace/issues/61

### CODE REVIEW

**Review by algattik — CHANGES_REQUESTED** [review]
In the spirit of the other ADRs, it would be useful to structure it as [Decision] and append a discussion, rather than discussion and then conclusion. This helps in the future finding out what was actually done

File: docs/developer/decision-records/2022-04-19-jwt-hardening/README.md [hunk 1]
```diff
@@ -0,0 +1,28 @@
+# JWT hardening
+
+[Json Web Tokens (JWT)](https://datatracker.ietf.org/doc/html/rfc7519) are used in EDC as means to authenticate IDS requests. The current implementation allows for a malicious entity that gets ahold of a JWT to impersonate the original sender, and thus send requests to any other participant. Two categories or impersonation attacks are possible depending on how the JWT used for the impersonation attack is obtained:
```
Comment (algattik) [inline docs/developer/decision-records/2022-04-19-jwt-hardening/README.md:3]:
```suggestion
[Json Web Tokens (JWT)](https://datatracker.ietf.org/doc/html/rfc7519) are used in EDC as means to authenticate IDS requests. The current implementation allows for a malicious entity that gets hold of a JWT to impersonate the original sender, and thus send requests to any other participant. Two categories or impersonation attacks are possible depending on how the JWT used for the impersonation attack is obtained:
```

File: docs/developer/decision-records/2022-04-19-jwt-hardening/README.md [hunk 2]
```diff
@@ -0,0 +1,28 @@
...
+
+1. "JWT reuse": A malicious participant might reuse JWTs sent to him as provider, to send requests to other participants as a consumer impersonating the signer of the JWT.
+1. "JWT leak": A malicious attacker that gets ahold of a leaked JWT token might use it to send request to any participant impersonating the signer of the JWT.
```
Comment (algattik) [inline docs/developer/decision-records/2022-04-19-jwt-hardening/README.md:6]: nit: different number styles used in same doc

File: docs/developer/decision-records/2022-04-19-jwt-hardening/README.md [hunk 3]
```diff
@@ -0,0 +1,28 @@
...
+
+As a first step we will implement a solution based on the second item, as this involves the least effort and bases on an already existing component in the current dataspace implementations (the registry).
+
+## JWT leak
+
+This is a less critical case as the first one, as it involves a previous JWT leak through a system security hole.
+
+DID domain matching (solution 3. in the previous section) might also solve this security issue in some scenarios. The DID of the consumer available as issuer of the JWT can be matched against the URL of the request sender. Nonetheless, resolving the origin URL can have complications as this can be manipulated by network components like gateways, leaving with no reliable way to get to this information.
+
+Initially we decide to go with no solution other than preventing JWT leaks themselves through common security best practices.
```
Comment (algattik) [inline docs/developer/decision-records/2022-04-19-jwt-hardening/README.md:28]: rephrase

Comment (marcgs) [reply docs/developer/decision-records/2022-04-19-jwt-hardening/README.md:34]: suggestion?

**Comment by marcgs** [comment]
> In the spirit of the other ADRs, it would be useful to structure it as [Decision] and append a discussion, rather than discussion and then conclusion. This helps in the future finding out what was actually done

Added decision section

File: docs/developer/decision-records/2022-04-19-jwt-hardening/README.md [hunk 4]
```diff
@@ -0,0 +1,34 @@
+# JWT hardening
```
Comment (cpeeyush) [inline docs/developer/decision-records/2022-04-19-jwt-hardening/README.md:1]: nit: Do we also want to include expiration claim? https://datatracker.ietf.org/doc/html/rfc7519#section-4.1.4

**Review by cpeeyush — APPROVED** [review]
Approved with suggestion.

File: docs/developer/decision-records/2022-04-19-jwt-hardening/README.md (hunk 4 above)
Comment (marcgs) [reply docs/developer/decision-records/2022-04-19-jwt-hardening/README.md:1]: This one is already set in EDC to 10 mins 👍🏻

**Review by algattik — APPROVED** [review]

**Pull request merged**